### ISSUE TITLE
Bug in rewriting with cubical primitives

### ISSUE THREAD

**Issue by FrozenWinters** [agda/agda#6418]
I am using Agda version 2.6.2.2 in Cubical mode

With the following header:

```agda
{-# OPTIONS --cubical --rewriting #-}

module Foo where

open import Agda.Primitive using (Level; lzero; lsuc; _⊔_) public
open import Cubical.Core.Everything public
open import Cubical.Foundations.Everything renaming (cong to ap) public

{-# BUILTIN REWRITE _≡_ #-}

private
  variable
    ℓ : Level
```

I am attempting to make the following rewrite:

```agda
compSides : {A : Type ℓ} (x : A) {y z : A} (p : y ≡ z)
  (i : I) (j : I) → Partial (i ∨ ~ i) A
compSides x p i =
  (λ where
    k (i = i0) → x
    k (i = i1) → p k)

lem1 : {A : Type ℓ} {x y z : A} (p : x ≡ y) (q : y ≡ z) (i : I) →
  hcomp
    (λ j .o → transp (λ _ → A) j (compSides x q i j o))
    (transp (λ _ → A) i0 (p i))
    ≡ (p ∙ q) i
lem1 {A = A} {x} p q i k =
  hcomp 
    (λ j .o → transp (λ _ → A) (j ∨ k) (compSides x q i j o))
    (transp (λ _ → A) k (p i))

{-# REWRITE lem1 #-}
```

This morally says that the following `badComp` function is the usual comp:

```agda
badComp : {A : Type ℓ} {x y z : A} → x ≡ y → y ≡ z → x ≡ z
badComp {A = A} {x} p q i =
  comp
    (λ j → A)
    (λ where
      j (i = i0) → x
      j (i = i1) → q j)
    (p i)
```

Upon loading the file, I get the error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__,
called at src/full/Agda/TypeChecking/Rewriting/NonLinPattern.hs:142:41
in Agd-2.6.2.2-cfc1fa6b:Agda.TypeChecking.Rewriting.NonLinPattern
```

**Comment by plt-amy** [comment]
Seems like this has been fixed, on ~~2.6.4~~ 120e34a3beb55b79feac3e295769bc17eab92f45 I get
```
/home/amelia/default/Projects/agda/Test40.agda:35,1-21
lem1  is not a legal rewrite rule, since the following variables are not bound by the left hand side:  p
when checking the pragma REWRITE lem1
```
which is.. not exactly what I was expecting, but definitely in line with what I was expecting, since matching the LHS of `lem1` looks _very_ hard to me. I'll try bisecting to find the fix, since it's probably in 2.6.3

**Comment by FrozenWinters** [comment]
Here is a simpler minimal program that demonstrates the bug:

```agda
compSides : {A : Type ℓ} (x : A) {y z : A} (p : y ≡ z)
  (i : I) (j : I) → Partial (i ∨ ~ i) A
compSides x p i =
  (λ where
    k (i = i0) → x
    k (i = i1) → p k)

lem2 : {A : Type ℓ} (x : A) (i : I) →
  hcomp (compSides x (refl {x = x}) i) (refl {x = x} i) ≡ x

{-# REWRITE lem2 #-}
```

This is trying to get `refl ∙ refl ≡ refl` to be a rewrite.
